crates/starknet/src/error.rs

```diff
@@ -47,6 +48,22 @@ pub enum Error {
     SerializationError { obj_name: String },
     #[error("Serialization not supported")]
     SerializationNotSupported,
+    #[error("{reason}")]
+    FeeError { reason: String },
+}
+
+#[derive(Debug, Error)]
+pub enum StateError {
+    #[error("No class hash {0} found")]
```

Comment:
not for all errors, because errors under the BlockifierTransactionExecutionError print the StarkFelt with debug formatter

Reply:
So it will print an array of decimal represenations of bytes? That's unforunate. How can we fix that (potentially in a future PR)?

Reply:
create our error type, or beg them to change the formatter

Reply:
Or I see now that their debug format is actually hex representation of the byte array

Reply:
So the issue might be closeable.

Instead of deciding on the destiny of #164 now, I think after merging this PR we should go through all the issues, especially ones mentioning starknet_in_rust, and check if they can be closed.

Reply:
ok